src/GafferScene/EditScopeAlgo.cpp

```diff
@@ -854,3 +855,210 @@ const Gaffer::GraphComponent *GafferScene::EditScopeAlgo::attributeEditReadOnlyR
+
+	PathFilterPtr addPathFilter = new PathFilter;
+	result->addChild( addPathFilter );
+	addPathFilter->pathsPlug()->setInput( spreadsheet->outPlug()->getChild<StringVectorDataPlug>( g_addSetColumnIndex ) );
+
+	PathFilterPtr removePathFilter = new PathFilter;
+	result->addChild( removePathFilter );
+	removePathFilter->pathsPlug()->setInput( spreadsheet->outPlug()->getChild<StringVectorDataPlug>( g_removeSetColumnIndex ) );
+
+	GafferScene::SetPtr addSet = new GafferScene::Set();
+	result->addChild( addSet );
+	addSet->inPlug()->setInput( result->inPlug() );
+	addSet->filterPlug()->setInput( addPathFilter->outPlug() );
+	addSet->namePlug()->setInput( spreadsheet->enabledRowNamesPlug() );
+	addSet->modePlug()->setValue( GafferScene::Set::Mode::Add );
+	addSet->setVariablePlug()->setValue( "setMembership:set" );
+
+	GafferScene::SetPtr removeSet = new GafferScene::Set();
+	result->addChild( removeSet );
+	removeSet->inPlug()->setInput( addSet->outPlug() );
+	removeSet->filterPlug()->setInput( removePathFilter->outPlug() );
+	removeSet->namePlug()->setInput( spreadsheet->enabledRowNamesPlug() );
+	removeSet->modePlug()->setValue( GafferScene::Set::Mode::Remove );
+	removeSet->setVariablePlug()->setValue( "setMembership:set" );
+
+	auto rowsPlug = static_cast<Spreadsheet::RowsPlug *>(
+		PlugAlgo::promoteWithName( spreadsheet->rowsPlug(), "edits" )
+	);
+
+	Metadata::registerValue( rowsPlug, "spreadsheet:defaultRowVisible", new BoolData( false ) );
+	Metadata::registerValue( rowsPlug->defaultRow(), "spreadsheet:rowNameWidth", new IntData( 300 ) );
+
+	result->outPlug()->setInput( removeSet->outPlug() );
+
+	return result;
+}
+
+EditScope::ProcessorRegistration g_setMembershipProcessorRegistration( "SetMembershipEdits", setMembershipProcessor );
+
+}  // namespace
+
+void EditScopeAlgo::setSetMembership( Gaffer::EditScope *scope, const std::string &set, const IECore::PathMatcher &paths, EditScopeAlgo::SetMembership state )
+{
+	Node *processor = scope->acquireProcessor( "SetMembershipEdits" );
+	auto rows = processor->getChild<Spreadsheet::RowsPlug>( "edits" );
+
+	auto row = rows->row( set );
+	if( !row )
+	{
+		row = rows->addRow();
+		row->namePlug()->setValue( set );
+	}
+
+	auto addCell = row->cellsPlug()->getChild<Spreadsheet::CellPlug>( g_addSetColumnIndex );
+	auto removeCell = row->cellsPlug()->getChild<Spreadsheet::CellPlug>( g_removeSetColumnIndex );
+
+	auto addStringPlug = addCell->valuePlug<StringVectorDataPlug>();
+	auto removeStringPlug = removeCell->valuePlug<StringVectorDataPlug>();
+
+	ConstStringVectorDataPtr addData = addStringPlug->getValue();
+	ConstStringVectorDataPtr removeData = removeStringPlug->getValue();
+
+	PathMatcher addPaths( addData->readable().begin(), addData->readable().end() );
+	PathMatcher removePaths( removeData->readable().begin(), removeData->readable().end() );
+
+	if( state == EditScopeAlgo::SetMembership::INHERIT )
+	{
+		if( !addPaths.removePaths( paths ) )
```

Comment:
Since both `add` and `remove` plugs are directly editable by the user as well, I think we'd be better off not assuming that a path is only in one plug or the other. So we should remove from both plugs here...

Reply:
Fixed in 70b56182b2eeb09a15f6b30c5c427ab16d26d12a